### PR TITLE
Assignment 3- parts 1 and 2

### PR DESCRIPTION
**Pull request by AustinHall7798**
I think I have everything, but any feedback is appreciated

### CODE REVIEW

File: Inheritance/test/Inheritance.Tests/PrinterTests.cs [hunk 1]
```diff
@@ -7,36 +7,101 @@
 namespace Inheritance.Tests
 {
     [TestClass]
-    public class PersonPrinterTests
+    public class PrinterTests
     {
         [TestMethod]
-        public void PersonGetsPrinted()
+        public void ItemGetsPrinted()
```
Comment (JerettLatimer) [inline Inheritance/test/Inheritance.Tests/PrinterTests.cs]: Good job on your tests. They effectively test for expected and edge cases including null.

**Comment by AustinHall7798** [comment]
After looking at others' assignments, I realize maybe I wasn't supposed to create a constructor for Food and Television so I'll go without that on the extra credit portion

File: Inheritance/test/Inheritance.Tests/ActorTests.cs [hunk 2]
```diff
@@ -6,8 +7,67 @@ namespace Inheritance.Tests
...
+        }
+
+        [TestMethod]
+        public void PennySpeakTest_()
+        {
+            //Arrange
+            Penny penny = new Penny();
+
+            //Act
+            string speech = ExtensionClass.Speak(penny);
```
Comment (breaman) [inline Inheritance/test/Inheritance.Tests/ActorTests.cs]: Since this was written as an extension method, you could have just called this as penny.Speak() as well.

File: Inheritance/test/Inheritance.Tests/ActorTests.cs [hunk 3]
```diff
@@ -6,8 +7,67 @@ namespace Inheritance.Tests
...
         [TestMethod]
-        public void TestMethod1()
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void NullValuePassedToSpeak_ExceptionExpected()
         {
+            ExtensionClass.Speak(null);
+        }
+
+        [TestMethod]
+        public void PennySpeakTest_()
```
Comment (breaman) [inline Inheritance/test/Inheritance.Tests/ActorTests.cs]: you shouldn't end your method names with an underscore.

File: Inheritance/test/Inheritance.Tests/ActorTests.cs [hunk 4]
```diff
@@ -6,8 +7,67 @@ namespace Inheritance.Tests
...
+        public void SheldonSpeakTest_()
+        {
+            //Arrange
+            Sheldon sheldon = new Sheldon();
+
+            //Act
+            string speech = ExtensionClass.Speak(sheldon);
+
+            //Assert
+            Assert.AreEqual(speech, "Bazinga!");
```
Comment (breaman) [inline Inheritance/test/Inheritance.Tests/ActorTests.cs]: the signature for AreEqual is (expected, actual) so these parameters should have been swapped.
